### PR TITLE
Favorites

### PR DESCRIPTION
**Pull request by AGS-**
Works as expected now. Let's cut a release after this.

### CODE REVIEW

File: src/components/HistoryQuery.js [hunk 1]
```diff
@@ -1,29 +1,88 @@
...
+      this.setState({
+        favorite: false,
+        starIcon: this.getStarIcon(false),
+      });
+    } else if (this.state.favorite === false) {
+      this.setState({
+        favorite: true,
+        starIcon: this.getStarIcon(true),
+      });
+    }
```
Comment (wincent) [inline src/components/HistoryQuery.js]: Seems this whole `if` could be replaced with something like this:

```
this.setState({
  favorite: !this.state.favorite,
  starIcon: this.getStarIcon(!this.state.favorite),
});
```

or similar.

File: src/components/QueryHistory.js [hunk 2]
```diff
@@ -39,28 +42,39 @@ export class QueryHistory extends React.Component {
...
-        operationName: nextProps.operationName || '',
-      });
-      if (this.store.length > MAX_HISTORY_LENGTH) {
-        this.store.shift();
+        variables: nextProps.variables,
+        operationName: nextProps.operationName,
+      };
+      // if (this.favoriteStore && this.favoriteStore.contains(item)) {
+      //   item.favorite = true;
+      // }
```
Comment (wincent) [inline src/components/QueryHistory.js]: Don't forget to deal with the commented out code.

File: src/components/QueryHistory.js [hunk 3]
```diff
@@ -86,4 +105,23 @@ export class QueryHistory extends React.Component {
       </div>
     );
   }
+
+  toggleFavorites = (query, variables, operationName) => {
```
Comment (wincent) [inline src/components/QueryHistory.js]: Probably `toggleFavorite`, right?

File: src/components/HistoryQuery.js [hunk 4]
```diff
@@ -1,29 +1,88 @@
...
-    onSelect(query, variables, operationName);
+export default class HistoryQuery extends React.Component {
+  static propTypes = {
+    query: PropTypes.string,
+    variables: PropTypes.string,
+    operationName: PropTypes.string,
+    favorite: PropTypes.bool,
+    favoriteSize: PropTypes.number,
+    onToggleFavorites: PropTypes.func,
+    onSelect: PropTypes.func,
```
Comment (wincent) [inline src/components/HistoryQuery.js]: nit: alphabetize

**Comment by wincent** [comment]
Screenshot might help frame the discussion of the UX here.

**Comment by AGS-** [comment]
screen shot 2017-06-22 at 5 28 52 pm

**Comment by AGS-** [comment]
graphiqlfavorites

**Pull request merged**